### PR TITLE
Change markdownlinter to default rules to false, and we have to add more in

### PR DESCRIPTION
**Pull request by ValarDragon**
## What is the purpose of the change

Closes: #2287 

Stops the time burn on things that don't matter, wrt most markdown lints. Most lints are things that all renderers handle, and aren't particularly obtrusive in any IDE.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not documented

### CODE REVIEW

File: .markdownlint.yml [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
-MD024: false
-MD029: false
-MD033: false
-MD037: false # breaks on latex
+MD018: true
+MD026: true
+MD034: true
+MD040: true
+MD042: true
+MD048: true
```
Comment (p0mvn) [inline .markdownlint.yml:11]: Link fragments should be valid

```suggestion
MD048: true
MD051: true
```

File: .markdownlint.yml [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
-MD007: false
-MD009: false
-MD010:
-  code_blocks: false
-MD013: false
-MD024: false
-MD029: false
-MD033: false
-MD037: false # breaks on latex
+MD018: true
```
Comment (p0mvn) [inline .markdownlint.yml:6]: - Heading levels should only increment by one level at a time

I think this helps to ensure that headings make sense / are consistent

```suggestion
MD001: true
MD018: true
```

**Review by p0mvn — APPROVED** [review]

File: .markdownlint.yml [hunk 3]
```diff
@@ -1,16 +1,22 @@
...
+MD026: true
+MD034: true
+MD040: true
+MD042: true
+MD048: true
+# MD004: false
+# # Can't disable MD007 :/
```
Comment (p0mvn) [inline .markdownlint.yml:13]: Are we keeping these just in case we want to switch back?

**Review by alexanderbez — APPROVED** [review]

File: .markdownlint.yml (hunk 3 above)
Comment (ValarDragon) [reply .markdownlint.yml:15]: I think so, we can def delete them later tho

**Pull request merged**